examples/rules.mk

```diff
@@ -168,9 +168,25 @@ else
 include $(OPENCM3_DIR)/mk/genlink-rules.mk
 endif
 
+# Find libopencm3 library folder
+#  this tries to match LIBNAME to manufacturer or manufacturer/series folder in lib/
+#  eg. LIBNAME=lm3s LIBFOLDER=lib/lm3s or LIBNAME=stm32f7 LIBFOLDER=lib/stm32/f7
+DIR=$(notdir $(subst /.,,$(wildcard $1/*/.)))
+LIBFOLDER=lib/$(strip \
+    $(foreach M,$(filter-out usb ethernet dispatch,$(call DIR,$(OPENCM3_DIR)/lib)), \
+        $(if $(subst opencm3_$M,,$(LIBNAME)),\
+            $(foreach S,$(call DIR,$(OPENCM3_DIR)/lib/$(M)), \
+                $(if $(subst opencm3_$M$S,,$(LIBNAME)),,$M/$S)), \
+            $M)))
+# exceptions
```

Comment:
definitely not interested in this one.  I've had enough rpoblems with people trying to pick and choose things, and reportings bugs when all they really needed to do was just run "make" once in the top and everythign would have been fine.  This relies far too much on knowledge of the build system and file structure within the library, which should be treated as a simple .a dependency.  It's a submodule for exactly that reason, just build it once, move on.